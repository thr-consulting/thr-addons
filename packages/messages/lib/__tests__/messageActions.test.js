import {
	addMessage,
	addSuccessMessage,
	addErrorMessage,
	addWarnMessage,
	clearMessage,
	clearMessages,
} from '../messagesActions';

const mockMath = Object.create(global.Math);
mockMath.random = () => 0.6920730585883441;
global.Math = mockMath;

const message = {
	id: 'myid',
	level: 'success',
	message: 'This is a message',
	title: 'This is a title',
};

describe('Message Actions', () => {
	it('should return a message with a random id', () => {
		const ret = addMessage(message.level, message.message, message.title);
		expect(ret).toMatchSnapshot();
	});

	it('should return a message with a set id', () => {
		const ret = addMessage(message.level, message.message, message.title, '1234');
		expect(ret).toMatchSnapshot();
	});

	it('should return a success message', () => {
		const ret = addSuccessMessage(message.message);
		expect(ret).toMatchSnapshot();
	});

	it('should return an error message', () => {
		const ret = addErrorMessage(message.message);
		expect(ret).toMatchSnapshot();
	});

	it('should return a warning message', () => {
		const ret = addWarnMessage(message.message);
		expect(ret).toMatchSnapshot();
	});

	it('should return a clear message action', () => {
		const ret = clearMessage('1234');
		expect(ret).toMatchSnapshot();
	});

	it('should return a clear all messages action', () => {
		const ret = clearMessages();
		expect(ret).toMatchSnapshot();
	});
});
