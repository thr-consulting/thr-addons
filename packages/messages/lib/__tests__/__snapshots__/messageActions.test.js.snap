// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Actions should return a clear all messages action 1`] = `
Object {
  "type": "CLEAR_MESSAGES",
}
`;

exports[`Message Actions should return a clear message action 1`] = `
Object {
  "id": "1234",
  "type": "CLEAR_MESSAGE",
}
`;

exports[`Message Actions should return a message with a random id 1`] = `
Object {
  "message": Object {
    "id": "d5mycmr0npclkde1m7vi",
    "level": "success",
    "message": "This is a message",
    "title": "This is a title",
  },
  "type": "ADD_MESSAGE",
}
`;

exports[`Message Actions should return a message with a set id 1`] = `
Object {
  "message": Object {
    "id": "1234",
    "level": "success",
    "message": "This is a message",
    "title": "This is a title",
  },
  "type": "ADD_MESSAGE",
}
`;

exports[`Message Actions should return a success message 1`] = `
Object {
  "message": Object {
    "id": "success",
    "level": "success",
    "message": "This is a message",
    "title": "Success",
  },
  "type": "ADD_MESSAGE",
}
`;

exports[`Message Actions should return a warning message 1`] = `
Object {
  "message": Object {
    "id": "warn",
    "level": "yellow",
    "message": "This is a message",
    "title": "Warning",
  },
  "type": "ADD_MESSAGE",
}
`;

exports[`Message Actions should return an error message 1`] = `
Object {
  "message": Object {
    "id": "error",
    "level": "error",
    "message": "This is a message",
    "title": "Error",
  },
  "type": "ADD_MESSAGE",
}
`;
